Modules/IO/JPEG/src/itkJPEGImageIO.cxx

```diff
@@ -36,7 +36,7 @@ extern "C"
   {
     /* cinfo->err really points to a itk_jpeg_error_mgr struct, so coerce pointer
      */
-    auto * myerr = reinterpret_cast<itk_jpeg_error_mgr *>(cinfo->err);
+    itk_jpeg_error_mgr * myerr = (itk_jpeg_error_mgr *)cinfo->err;
```

Comment:
I have imagined JPEG style here. OK.

Reply:
I have reverted. But still don't know, IMHO C code were not so bad in `extern "C"` block.

Reply:
I would prefer to use C-style here like in all implementation i have seen. The function is a kind of extension of JPEG lib. From Wikipedia: `Some compilers will produce non-working code if a function pointer declared extern "C" points to a C++ function that is not declared extern "C".` There is also related discussion [Does a C++ cast strip the 'extern "C"' from a declaration?](https://stackoverflow.com/questions/39584003/does-a-c-cast-strip-the-extern-c-from-a-declaration). Honestly i am too lazy to read that discussion, IMHO, it is safer to avoid "low reward" experiments here.